src/core/tsi/ssl_transport_security.cc

```diff
@@ -344,23 +344,35 @@ static tsi_result add_subject_alt_names_properties_to_peer(
     size_t subject_alt_name_count) {
   size_t i;
   tsi_result result = TSI_OK;
-
   /* Reset for DNS entries filtering. */
   peer->property_count -= subject_alt_name_count;
-
   for (i = 0; i < subject_alt_name_count; i++) {
     GENERAL_NAME* subject_alt_name =
         sk_GENERAL_NAME_value(subject_alt_names, TSI_SIZE_AS_SIZE(i));
-    /* Filter out the non-dns entries names. */
-    if (subject_alt_name->type == GEN_DNS) {
+    if (subject_alt_name->type == GEN_DNS ||
+        subject_alt_name->type == GEN_EMAIL ||
+        subject_alt_name->type == GEN_URI) {
       unsigned char* name = nullptr;
       int name_size;
-      name_size = ASN1_STRING_to_UTF8(&name, subject_alt_name->d.dNSName);
+      if (subject_alt_name->type == GEN_DNS) {
+        name_size = ASN1_STRING_to_UTF8(&name, subject_alt_name->d.dNSName);
+      } else if (subject_alt_name->type == GEN_EMAIL) {
+        name_size = ASN1_STRING_to_UTF8(&name, subject_alt_name->d.rfc822Name);
+      } else {
+        name_size = ASN1_STRING_to_UTF8(
+            &name, subject_alt_name->d.uniformResourceIdentifier);
+      }
       if (name_size < 0) {
         gpr_log(GPR_ERROR, "Could not get utf8 from asn1 string.");
         result = TSI_INTERNAL_ERROR;
         break;
       }
+      if (subject_alt_name->type == GEN_URI) {
+        result = tsi_construct_string_peer_property(
```

Comment:
Just wondering why you want to store URI SAN twice in peer properties.

Reply:
Nice catch. I think that's the left-over from previous implementation. The code is revised so that URI SAN field is added only once.